library/core/src/ptr/mod.rs

```diff
@@ -57,50 +57,44 @@
 //! [`NonNull::dangling`] in such cases.
 //!
 //! ## Pointer to reference conversion
-//! When converting a pointer to a reference `&T` using `&*`,
+//!
+//! When converting a pointer to a reference (e.g. via `&*ptr` or `&mut *ptr`),
 //! there are several rules that must be followed:
 //!
 //! * The pointer must be properly aligned.
 //!
-// some microprocessors may use address 0 for an interrupt vector.
-// users of these microprocessors must always read/write address 0 through
-// a raw pointer, not a reference.
```

Comment:
I'm not sure what the point of this was, it was anyway hidden from users. If we want to give guidance on how to access address 0, this is the wrong place. (I think currently it is always UB to access address 0, even with raw pointers.)